World.py

```diff
@@ -108,7 +110,61 @@ def __init__(self, id, settings):
 
         self.resolve_random_settings()
 
+        if len(settings.hint_dist_user) == 0:
+            dists_json = os.listdir(data_path('Hints/'))
+            for d in dists_json:
+                dist = read_json(os.path.join(data_path('Hints/'), d))
+                if dist['name'] == self.hint_dist:
+                    self.hint_dist_user = dist
+        else:
+            self.hint_dist = 'custom'
+            
+        # Validate hint distribution format
+        # Originally built when I was just adding the type distributions
+        # Location/Item Additions and Overrides are not validated
+        hint_dist_valid = False
+        if all(key in self.hint_dist_user['distribution'] for key in hint_dist_keys):
+            hint_dist_valid = True
+            sub_keys = {'order', 'weight', 'fixed', 'stones'}
+            for key in self.hint_dist_user['distribution']:
+                if not all(sub_key in sub_keys for sub_key in self.hint_dist_user['distribution'][key]):
+                    hint_dist_valid = False
+        if not hint_dist_valid:
+            raise InvalidFileException("Hint distributions require all hint types be present in the distro (trial, always, woth, barren, item, song, overworld, dungeon, entrance, sometimes, random, junk, named-item). If a hint type should not be shuffled, set its order to 0. Hint type format is \"type\": { \"order\": 0, \"weight\": 0.0, \"fixed\": 0, \"stones\": 0 }")
+        
+        self.added_hint_types = {}
+        self.item_added_hint_types = {}
+        self.hint_type_overrides = {}
+        self.item_hint_type_overrides = {}
+        for dist in hint_dist_keys:
+            self.added_hint_types[dist] = []
+            for loc in self.hint_dist_user['add_locations']:
+                if 'types' in loc:
+                    if dist in loc['types']:
+                        self.added_hint_types[dist].append(loc['location'])
+            self.item_added_hint_types[dist] = []
+            for i in self.hint_dist_user['add_items']:
+                if dist in i['types']:
+                    self.item_added_hint_types[dist].append(i['item'])
+            self.hint_type_overrides[dist] = []
+            for loc in self.hint_dist_user['remove_locations']:
+                if dist in loc['types']:
+                    self.hint_type_overrides[dist].append(loc['location'])
+            self.item_hint_type_overrides[dist] = []
+            for i in self.hint_dist_user['remove_items']:
+                if dist in i['types']:
+                    self.item_hint_type_overrides[dist].append(i['item'])
+
+        self.hint_text_overrides = {}
+        for loc in self.hint_dist_user['add_locations']:
+            if 'text' in loc:
+                # Arbitrarily throw an error at 80 characters to prevent overfilling the text box.
```

Comment:
There's a thought. Is there actually a specific limit?

Reply:
I'm not sure. Max length in my new hints is 71 characters for `Spirit Temple MQ Map Room Enemy Chest`. I tested it in-game with a Triforce Piece in it, and there was plenty of room left visually. 80 seemed like a decent cutoff.